applications/minotari_app_utilities/src/network_check.rs

```diff
@@ -52,15 +52,26 @@ pub const TARGET_NETWORK: Target = Target::NextNet;
 #[cfg(all(not(tari_network_mainnet), not(tari_network_nextnet)))]
 pub const TARGET_NETWORK: Target = Target::TestNet;
 
-pub fn is_network_choice_valid(network: Network) -> Result<(), NetworkCheckError> {
+pub fn is_network_choice_valid(network: Network) -> Result<Network, NetworkCheckError> {
     match (TARGET_NETWORK, network) {
-        (Target::MainNet, Network::MainNet | Network::StageNet) => Ok(()),
+        (Target::MainNet, n @ Network::MainNet | n @ Network::StageNet) => Ok(n),
         (Target::MainNet, _) => Err(NetworkCheckError::MainNetBinary(network)),
 
-        (Target::NextNet, Network::NextNet) => Ok(()),
+        (Target::NextNet, n @ Network::NextNet) => Ok(n),
         (Target::NextNet, _) => Err(NetworkCheckError::NextNetBinary(network)),
 
-        (Target::TestNet, Network::LocalNet | Network::Igor | Network::Esmeralda) => Ok(()),
+        (Target::TestNet, n @ Network::LocalNet | n @ Network::Igor | n @ Network::Esmeralda) => Ok(n),
         (Target::TestNet, _) => Err(NetworkCheckError::TestNetBinary(network)),
     }
 }
+
+pub fn set_network_if_choice_valid(network: Network) -> Result<(), NetworkCheckError> {
+    match is_network_choice_valid(network) {
+        Ok(network) => {
+            let mut current_network = CURRENT_NETWORK.lock().unwrap();
```

Comment:
It seems [unlikely](https://doc.rust-lang.org/std/sync/struct.Mutex.html#panics), but since this could panic, it might be safer to propagate errors from `lock()` since you already return a `Result`.

Reply:
Fixed